### ISSUE TITLE
[Due for payment 2025-03-17] Use persistent bottom tabs in Settings to improve Workspace Editor UX

### ISSUE THREAD

**Issue by shawnborton** [Expensify/App#44242]
We're currently displaying Workspace Settings in the equivalent of a full screen modal. This means you can't easily navigate between Inbox, Search, and Workspace Settings - each time you close the Workspace Editor, you need to navigate back to Settings > Workspaces > Launch the workspace editor. 

Let's make it so that the bottom tab bar in Settings is persistent or sticky, in that even after you navigate one or two levels deeper, it remains:

https://github.com/Expensify/App/assets/2319350/6925987c-3b44-4ec8-a391-7d94a502c49d

![CleanShot 2024-06-24 at 12 35 20@2x](https://github.com/Expensify/App/assets/2319350/d5074086-026f-4026-823b-f226a476b344)

Other notes for completeness:
- We'll want to reshuffle the ordering of the LHN items for Settings so that Workspaces is higher up on the list and more discoverable (and always above the fold)
- We don't want to modify any of the existing native app back button behavior (this came up in Slack)
- If you are navigated a level or more deep within Settings, tapping on the Settings tab in the bottom tab bar should bring you back to the top level of Settings

Figma for this is [here](https://www.figma.com/design/hMyXUbmDI7ooMq4ygVNaNX/Wave-8%3A-Design-Updates-to-Ideal-Nav-Structure-(%23ideal-nav)?node-id=349-41917&t=zMuQ88uO4KboLz6I-1)

cc @Expensify/design @mountiny 

<details><summary>Issue Owner</summary>Current Issue Owner: @CortneyOfstad</details>

**Comment by mountiny** [comment]
Waiting for updates from SWM

**Comment by mountiny** [comment]
Asked for an update [here](https://expensify.slack.com/archives/C036QM0SLJK/p1719913162010719)

**Comment by mountiny** [comment]
Will follow up with the team

**Comment by mountiny** [comment]
Asked here https://expensify.slack.com/archives/C04878MDF34/p1720655134137259

**Comment by WojtekBoman** [comment]
Hi! Sorry this took a while, but there was a lot of navigation issues recently, and we needed some time to clean things up.
We investigated this topic together with Adam. Here are our thoughts:

The main change that we need to introduce is the possibility to display BottomTab conditionally on every page in the app, not only within the BottomTabNavigator (the way it's currently done). This approach requires some deeper changes to the current navigation architecture and may cause issues.

The first visible issue is connected with displaying bottom tab component several times in several navigators. Currently, it's rendered once in the BottomTabNavigator without any conditions, but to solve this issue we'd need to display it depending on the page and screen width not navigator (its position isn't would no longer be fixed). It may cause small flickering of the bottom tab during transition between screens as the bottom tab contains the floating action button which has very complex logic.

Another problem is the inconsistency of the animations. Currently, when we navigate between screens with the bottom tab, we don't display any animation, and when we navigate from the bottom tab to the central pane, we have a slide animation. Now central pane screen can have a bottom tab, so sometimes when navigating to the CentralPane we would show an animation and other times not. This is presented in the video below.

So to sum up: as I mentioned earlier, the display of BottomTab on any screen in the app is related to the refactor of the navigation architecture. In the current structure we use BottomTabNavigator to display the bottom bar, now it would no longer be the responsibility of this navigator, so we will have to develop a new approach to handle display of the bottom tab.

These are the results of our research regarding this functionality, What do you think about these considerations? would you like us to continue working on this?


https://github.com/Expensify/App/assets/47774969/65dfd68d-89f3-4052-ad3e-76c706857eeb

**Comment by mountiny** [comment]
@WojtekBoman Thanks for the write up, I still think we want to explore this path more.

I believe its common to have the bottom tab on more pages in the app. It kinda felt like our solution is a bit custom before and what we are proposing here is more in line with standard usage of bottom tab navigator

**Comment by shawnborton** [comment]
Totally agree that we want to explore this further, especially given that we think we will want to reuse this pattern for Domains as well.

**Comment by Kicu** [comment]
@shawnborton @mountiny we will continue with the research to bring some more specific info. That is mostly done by @adamgrzybowski and @WojtekBoman but I'm sometimes helping a bit.

For now a question to get a bit more specifics:
Do you guys see any more views at this point where you would like to see Bottom Tab? (asking because the issue started with Workspaces Page). Because I believe adding a bottom tab on _every_ Page is not something we want? (chats? settings?)

The more requirements we know right now - the better the solution will be, and navigation in this app really is quite complex ;) Every time we want to do some change we need to think about: mobile native apps, wide layout and how would that affect URL/route for browsers.

**Comment by shawnborton** [comment]
Right now I think we're really only imagining the Settings > Workspaces > Workspace Editor, as well as (soon to be) Settings > Domains > Domain Editor. That being said, I could totally see an argument where we might as well do this for all Settings pages to be consistent, but I don't know if that is really a requirement or not. cc @Expensify/design for any additional thoughts too.

**Comment by dannymcclain** [comment]
No additional thoughts from me, I agree with your summary!

**Comment by JmillsExpensify** [comment]
I don't think it's required for the other pages, like profile or security and such, but I do think we should do this for workspaces (and domains in the future). I don't think we need to do it in other places.

**Comment by mountiny** [comment]
@Kicu @adamgrzybowski Do you have any further questions to get you unblocked? Thanks!

**Comment by Kicu** [comment]
Hey, we synced with guys and discussed some more details.

**First to re-iterate:**
The main difficulty in our navigation is that we support both narrow and wide layouts and we expect the navigation to be consistent between the two. That means having a lot of code handling screens on narrow + bigger devices and transitioning from one to the other. In order to introduce the bottom bar in more screens we will have to modify said logic.


We would like to ask if it's possible for the UI/UX people to give us 2-3 examples of apps where navigation and bottom bar behaves consistently and similar to what we would like to have in Expensify in future?
(_Some examples of desktop apps (wide layout) would also be nice, but these are harder to find._)
Looking at such apps would help us find any possible edge cases and understand different navigation flows better.

We have tried to search for desktop+mobile apps ourselves, but couldn't find any desktop apps that behave similar to ours. I have checked Spotify, Slack and Notion - not the deepest search but at least these are popular.

Some insights:
 - popular mobile apps have bottom bar on some views but not on all of them, like you would expect...
 - **however** desktop apps very often don't have bottom bar, even if their mobile versions has it
 - mobile slack shows the bottom tab on different views on ios and android. For example slack chat on iphone will have the bottom bar and on android it will not - I find that interesting but dunno why it's like that;

To sum up, going back to this quote from @mountiny 
 > [...] our solution is a bit custom before and what we are proposing here is more in line with standard usage of bottom tab navigator

I agree with this in regards to mobile apps, but when talking about mobile+wide layout things are not so clear cut 😉

Now for something less serious, here are some first experiments that @WojtekBoman did for the bottom bar 😅 

https://github.com/user-attachments/assets/3afad7c2-c4a0-4795-b23e-c9fd2a689c27

^ I know this is silly, but it shows exactly the kinds of edge cases that we are ironing out right now, to avoid issues later.

**Comment by adamgrzybowski** [comment]
> I agree with this in regards to mobile apps, but when talking about mobile+wide layout things are not so clear cut 😉

Yes, this is an important point. I'd say we have a custom solution for the bottom tabs anyway. This does not depend on the number of screens that do or do not have a bottom tab bar.

What is custom and new though is that we want to introduce a conditional bottom tab depending on the screen size.

**Comment by dannymcclain** [comment]
Happy to try to look around a bit, but FWIW I think it's going to be pretty hard to find any other apps doing what we're doing. The way we try to maintain consistency across platforms is really not normal, so I suspect it will be difficult to find examples to draw inspiration from. But I'll try to take a look regardless!

**Comment by shawnborton** [comment]
Yeah, that's my general sentiment too. But to clarify, even if we don't find a perfect example, we can still make progress on this, right?

**Comment by mountiny** [comment]
Yeah, I agree. The fact our app is trying to maintain consistency pretty hard means that there are not many examples like this. This makes it a bit tricky to handle, I think what could help is if the design team created some interactive mockups with various flows to catch all the edge cases and establish the patterns we want in this updated form of navigation. Would that help to get a better understanding of what the expected output is?

**Comment by shawnborton** [comment]
Happy to make more mockups and flows if we think the original screenshots/video from the first comment above won't suffice. It's hard to think of any "edge cases" right now since we really only want to apply this to the Workspaces/Domains page.

**Comment by mountiny** [comment]
Maybe we might want to try to mocking some flow where the user resizes the window from wide to narrow, takes some further actions, and resizes back to wide. I know most people does not do that but we need to make sure the app navigation does not break when this happens

**Comment by shawnborton** [comment]
Cool, happy to do that. Let's see what SWM says they need in terms of more mocks/flows as well.

**Comment by adamgrzybowski** [comment]
I talked with guys and we came up with an idea. The base of requirements sounds quite simple. By the base I mean:

**1. These screens have a bottom tab bar:**
- Home
- Search bottom tab screen
- Settings Initial
- Workspaces initial

**2. These screens have a bottom tab bar but only on small screen**
- Workspaces

I am not sure if we want Profile, Members, and other screens from the Workspace settings to have it. Please verify what is expected.

**3. If you are navigated a level or more deep within Settings, tapping on the Settings tab in the bottom tab bar should bring you back to the top level of Settings**

The last one is a little vague: 

**4. We don't want to modify any of the existing native app back button behavior (this came up in Slack)** 

Instead of theorizing which use cases can break, we will prepare a POC of points 1 to 3 and won't modify any other navigation behavior. Then you will be able to play around with the POC and create a list of cases where the navigation feels off. We assume that there will be at least a few cases like that. 

I think it will be much easier to talk about the next steps when everybody can test and see examples instead of drawing screens with some arrows and comments.

**Comment by shawnborton** [comment]
> 2. These screens have a bottom tab bar but only on small screen
> I am not sure if we want Profile, Members, and other screens from the Workspace settings to have it. Please verify what is expected.

We just want this for Workspaces to start. 


> 3. If you are navigated a level or more deep within Settings, tapping on the Settings tab in the bottom tab bar should bring you back to the top level of Settings
> The last one is a little vague:

The idea is that if you are in Settings > Workspaces > My Workspace, and you tap on the Settings button in the bottom nav bar, it will bring you all the way back to the top of the hierarchy to Settings. Let me know if that makes sense. It's kind of like a common behavior you see in other apps where pressing the bottom tab again in a page that is active might scroll you to the top or something. 

Love the idea of a POC, excited to play with it!

**Comment by mountiny** [comment]
@WojtekBoman @Kicu What is the latest on this one?

**Comment by mountiny** [comment]
Thanks for the promising update Adam

**Comment by adamgrzybowski** [comment]
👋 Hey guys!
I thought it may be worth briefly describing the POC I'm currently working on 🛠️ 

Currently, our root navigator is the one handling the split view. The first route is always the `BottomTabNavigator` containing all sidebar screens.  

This may be problematic if we want to navigate to the bottom tab screen from a central pane screen on a **narrow layout**. We have to pop all other screens from the root navigator to uncover the first route. 

The idea is to create many smaller split navigators that we can push on the root. There will be a split navigator for each of the tabs:
- ReportsNavigator
- SearchNavigator
- SettingsNavigator
- WorkspacesNavigator (It doesn't have to be a separate button than the one for settings)

Now we will be able to push one split navigator on another one and we don't have to pop anything

This and the conditionally rendered bottom tab should be a solution for our issue.

There will be other benefits from these changes 🪨 🤾 🐦‍⬛ 🐦‍⬛ 
- Root navigator won't have the split responsibilities anymore. This is great because sometimes resizing can change the root state key and this can break the history.

- Many split navigators grouped by tab and related functionality should be easier to understand and maintain than one big root navigator.

- Because of separation, we prevent a whole category of bugs: sidebar and central pane out of sync. 

- The split navigator will only have a sidebar screen and central pane screens, it will be easier to implement for native platforms.

I know the last one may be especially important given how [this issue](https://github.com/Expensify/App/issues/35854) is going. 

Excited to work on this issue! Back to coding! 👨‍💻 


cc: @mountiny @WojtekBoman

**Comment by mountiny** [comment]
> 🪨 🤾 🐦‍⬛ 🐦‍⬛

That sounds very promising! Thanks for the write up

**Comment by shawnborton** [comment]
What's the latest on this one? cc @JmillsExpensify

**Comment by adamgrzybowski** [comment]
Hello everyone! An update from me.

As mentioned earlier, the goal we want to achieve requires changes in navigation architecture. 

I know that persistent bottom tabs don't sound like a big feature so it should be quick to implement but please keep in mind that these changes are much deeper and will make navigation simpler and more reliable. It should also fix at least a few of the old problems that would be unsolvable otherwise.

Thanks in advance for your patience 🙇 and back to the update: 

In the POC I am working on I managed to replace BottomTabNavigator and central pane screens with many split navigators. Those are quite big changes but the app handled it pretty well.

### Changes mentioned above require modifications in some places related to navigation

__linkTo:__ This function has been a pain for quite a long time. I rewritten it to do 90% of its previous functionality with 4x less code. Simplicity should help us with keeping this part bug-free. Needs a bit more work but not much. 

__SearchPage:__ With the previous approach we had to have a separate search page for the bottom tab and central pane. It was tricky to handle and fast resizing could lead to bugs. Now we can do it with just one page. I would say it's 80% refactored.

__Workspace switcher:__ Works more or less for the search page and the part for reports is still to be done.

__getAdaptedSatet:__
This is mainly cleaning the code that won't be necessary anymore. But there will be some adjustments.

__goBack:__
This function is a pain as well. Implementation of many SplitNavigators should finally allow us to refactor it. 

__Navigation.ts:__
Similarly to the getAdaptedState, mainly cleaning and some adjustments.

FYI: I am going to be ooo for two next weeks. I will have some time on Monday though and I will pass my work to @WojtekBoman. 

cc: @mountiny @shawnborton

**Comment by mountiny** [comment]
Amazing, thanks for the update @adamgrzybowski

**Comment by JmillsExpensify** [comment]
Agreed, thanks a ton for working on this one. Great update!

**Comment by mountiny** [comment]
@WojtekBoman @Kicu Any updates for this week even if Adam was out?

**Comment by Kicu** [comment]
None from me, right now I'm mostly working with Search and if there are search-related nav issues.

**Comment by WojtekBoman** [comment]
Hey @vit, I recently started working further on POC, last week I was involved in topics related to OldDot Rules migration. Now I'm focused on connecting the new navigation logic with storing the policyID.

**Comment by adamgrzybowski** [comment]
I am back from ooo and I will continue my work on this issue together with @WojtekBoman

**Comment by mountiny** [comment]
Not overdue

**Comment by adamgrzybowski** [comment]
Hi everyone 👋  An update from us.

### Changes visible to the user 👁️ 
From visible changes, we implemented @shawnborton requirements regarding bottom tab behavior.

https://github.com/user-attachments/assets/467a8f14-25be-4d05-866a-7ff9bbd7b646


https://github.com/user-attachments/assets/141e376c-48bf-4842-9976-844dbe062729




Besides we found two regressions that we also fixed.

#### Regression 1: Policy should be changed to all if the chat doesn't belong to it.
Before:

https://github.com/user-attachments/assets/4d58e9c0-0ec3-4cce-aef5-5e322fc2fd7f

After:

https://github.com/user-attachments/assets/6c705e55-09c5-4a44-b678-355d6c4a7b43


#### Regression 2: Some really weird behavior of search when resizing.
Before:

https://github.com/user-attachments/assets/5d7b1d3c-d930-40e1-bd5f-4b7166473d62

After:

https://github.com/user-attachments/assets/9102f1b8-b5f5-460b-891b-895382c192dc

### Changes under the hood 🫣
However, what changed the most was the navigation structure. We know that the current structure is prone to something I would call the butterfly effect. Small changes in one place can cause regressions in others. My guess is that's how the regressions above were introduced.

That's why we are working on simplifying and bulletproofing navigation. 

Here you can see an example of the old linkTo file that grew over time and was the root of many bugs.

<img width="410" alt="image" src="https://github.com/user-attachments/assets/ba208836-526a-4f55-91e0-84c293074cc8">

And here's the new one.

<img width="541" alt="image" src="https://github.com/user-attachments/assets/a86938e4-60a1-40bc-97ca-5a5afe30b25d">

We believe this approach can help us catch bugs that have not yet been reported and fit well with ongoing efforts focused on quality.

### Testing
There are still things we need to adjust, but we think it's a good time to start testing the new navigators. We are removing a lot of complex code so we are expecting some regressions that we need to catch and fix.


#### Strategy
We talked about testing strategy and we want to utilize two ideas.

__The first one__
As in previous iterations, navigate around the app to see if the app feels right. We can already start this one.

__The second one__
Prepare more structured testing instructions.

We did notice that a lot of information about how navigation should work and what are the requirements is scattered somewhere over slack convos or in the closed issues.

We want to prepare a list of the testing steps covering all possible use cases for navigation separated into sections e.g. bottom tabs, go back.

This will be our list of requirements and testing steps at the same time. The idea is to extract them from already solved navigation issues.

Please let us know what you think about this approach and if you have an idea of what would be a good place to store the testing instructions for the navigation. We were thinking about the navigation documentation in the repo. Thanks!

cc: @mountiny @WojtekBoman 

P.S. At some point we switched to a different branch so here's the new draft PR https://github.com/Expensify/App/pull/49539

**Comment by dannymcclain** [comment]
That is so sick 🥹. This is coming along really well (at least from my perspective!)

>Regression 1: Policy should be changed to all if the chat doesn't belong to it.

For this one, I think I was under the impression that if you filtered by a workspace, we'd only surface results that were in that workspace. Am I wrong about that? Can a smart person weigh in there? (If I'm wrong about that assumption then I think the fix for that makes total sense.)

**Comment by mountiny** [comment]
@dannymcclain i dont think so, at least not in the current state of the search we do not consider what workspace you have selected

So that would be a new feature afaik and the above mentioned expected behaviour is correct

@adamgrzybowski @WojtekBoman great job on this one, i think its looking great and having the navigation structure simplified is a welcomed bonus. Regarding the test steps, is that something you want to handle compose or you would like us to create? There are some testing steps already from the navigation refactor doc, they should be in the doc and we can base off them. Asking as it was not clear to me who is supposed to work on that 😂

**Comment by mountiny** [comment]
I will try to trigger a build on the PR and start testing

**Comment by trjExpensify** [comment]
> @dannymcclain i dont think so, at least not in the current state of the search we do not consider what workspace you have selected

The workspace switcher should inform the search page, like it does the inbox page. CC: @luacmartins

**Comment by mountiny** [comment]
Good to know, thanks for chiming in

**Comment by dannymcclain** [comment]
>@dannymcclain i dont think so, at least not in the current state of the search we do not consider what workspace you have selected
>
>So that would be a new feature afaik and the above mentioned expected behaviour is correct

Ok cool, thanks!

>The workspace switcher should inform the search page, like it does the inbox page.

Sorry I wasn't talking about the search page, I was talking about the "Find" LHP—those are still two separate things right? 😅

**Comment by adamgrzybowski** [comment]
@mountiny I think you can play around with the app to find anything that looks weird. We tried this approach before and you helped us a lot 😄 @wojtek and I might miss some obvious bugs after looking at the new branch for so long.

For the structured testing steps I think we will create them and consult with you if they describe your requirements properly. That may help us find any miscommunication about how things should work.

After confirming we will add them to the docs.

BTW About the chat finder @dannymcclain I think we came to a conclusion long ago in some conversation that it would make more sense to allow the user to navigate to a chat outside of their workspace and switch to the "ALL" workspace than hide these chats and block the user. But I am not 100% 😄 Sounds like a perfect example of why we want to write down all these use cases in one place.

**Comment by mountiny** [comment]
> I think we came to a conclusion long ago in some conversation that it would make more sense to allow the user to navigate to a chat outside of their workspace and switch to the "ALL" workspace than hide these chats and block the user.

I believe that is correct, at least that is how we designed this before.

Sounds great, I am using the web app now and it is working well so far!

**Comment by adamgrzybowski** [comment]
Cool! In that case, we will continue adjusting the old code to the new structure and will send you the first batch of testing steps/requirements to confirm soon.

**Comment by adamgrzybowski** [comment]
Hi! Here is a short update from our side. Recently @WojtekBoman and I had to help with some other navigation issues so progress on persistent bottom tabs wasn't big. 

We continued our work on two parts requiring adjustment before testing and improving performance. 

- goBack <- We are halfway finished. It looks like we can finally get this function right without code spaghettification 🤞  

- getAdaptedState <- Also halfway finished

**Comment by trjExpensify** [comment]
>without code spaghettification 🤞

Actually lol'd at that 😆

**Comment by adamgrzybowski** [comment]
Hi guys! A quick update from us.

__getAdaptedSatet__
I pushed changes with the new getAdaptedState. Tomorrow I will test it more and look for regressions. There may be some because it handles a lot of cases but it seems pretty stable so far. 

__goBack__
While working on the goBack function, we looked through the archives📜 of the design docs for the first navigation refactor. It described two similar actions: go back and go up. 

Currently, the goBack function handles both cases by looking at the fallbackUrl and a few other indicators. This approach may be a bit outdated after so many changes to the nav structure since then. Also, the coupling of these two actions in one function is why it is so difficult to refactor. We are considering splitting it into two separate functions: goBack and goUp as a final API for other contributors.

I wonder if you could take a look at the goUp and goBack documentation and tell us what API you would expect as a developer after reading it. Your feedback would be very valuable 🙇

I found the docs [here](https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE/edit#heading=h.ns12dr1f35rx) but not sure if you all have access. 

A summary of the differences is described [here](https://github.com/Expensify/App/blob/main/contributingGuides/NAVIGATION.md#terminology)

__performance__
As you can see, performance is well... 😄 far from ideal. We identified two main reasons.

1. We don't have optimization for multiple split navigators in the stack.

2. Now we render the bottom tab bar per page and it's much more expensive than we initially thought.

We have a solid idea for the first problem and a few loose ones to check for the second problem, but as mentioned before, we want to take care of performance at the very end.

**Comment by JmillsExpensify** [comment]
P.S. @marcaaron curious for your thoughts, as you were involved in the original go back and go up implementation.

**Comment by mountiny** [comment]
> I wonder if you could take a look at the goUp and goBack documentation and tell us what API you would expect as a developer after reading it. Your feedback would be very valuable 🙇

Yep, I think the main difference between Up and Back is considering a deep link to some deeper lever of the stack. Consider you deep link from chat to settings/profile/address/country, and then you click the chevron UP. You go to the address page -> one page up in the stack, even if it was not technically in the browser history. If you go back after deep linking to that page, you just go back to the chat page where you came from.

I guess the confusion is that most of the time, both actions/ buttons do the same. Normally, you navigate through the stack, so going up also means going back in the browser history.

And as such, I am not sure if we need to divide these up to two different API actions for the developer, I kind of feel like the developer should not worry about it much

**Comment by mountiny** [comment]
As long as the documentation is clear, we can name is goUp, but I think it makes more sense in stack. I am sure there are cases where goBack on the other hand makes more sense

**Comment by WojtekBoman** [comment]
Hey, yesterday I published an update from me and @adamgrzybowski in our PR. To increase the visibility of the post, I am posting a link to it [here](https://github.com/Expensify/App/pull/49539#issuecomment-2402736884) :)

**Comment by mountiny** [comment]
This is still in progress

**Comment by mountiny** [comment]
Getting through the testing feedback

**Comment by mountiny** [comment]
ore great progress on the PR

**Comment by mountiny** [comment]
progressing

**Comment by mountiny** [comment]
Native stack about to be merged

**Comment by mountiny** [comment]
in progress

**Comment by mountiny** [comment]
PR is getting close, making this daily now

**Comment by mountiny** [comment]
All the bugs from the first QA round should be fixed, that did not include hybrid app as we did not have working ad hoc builds yet. I have now asked QA for another round including hybrid and hopefully that is the last one before we can merge this

**Comment by mountiny** [comment]
PR was merged, lets pray now

**Comment by mountiny** [comment]
Looking good, should be deploying to production soon

**Comment by pac-guerreiro** [comment]
@mountiny I noticed that the DebugTabView is not showing when debug mode is enabled. I did some digging and it seems that with the new bottom bar changes, we missed updating some names:

<img width="784" alt="Image" src="https://github.com/user-attachments/assets/7d5eb9c0-5005-49fc-9089-d1c286ef1ef6" />

If you want I can create a follow up PR with the fix!

**Comment by mountiny** [comment]
@pac-guerreiro yes please, raise a PR when you can!

**Comment by pac-guerreiro** [comment]
@mountiny [here](https://github.com/Expensify/App/pull/57772) it is 😄

**Comment by CortneyOfstad** [comment]
## Payment Summary

### NEWDOT
@shubham1206agra — to be paid $250 via NewDot
@dukenv0307 — to be paid $250 via NewDot

### CONTRACTED
@Kicu
@WojtekBoman 
@adamgrzybowski

**Comment by mountiny** [comment]
Agreed this was a large PR, I suggest $750 to both @shubham1206agra and @dukenv0307 

@CortneyOfstad could you please update the summary?

**Comment by mountiny** [comment]
I think those are follow ups as regressions - I am willing to go to $1000 but I think more would be a stretch

**Comment by CortneyOfstad** [comment]
Updated Summary found here — https://github.com/Expensify/App/issues/44242#issuecomment-2733356540

**Comment by CortneyOfstad** [comment]
@mountiny — is this completed or is anything else needed besides the updated payment summary? Thanks!

**Comment by mountiny** [comment]
this is completed, we can pay and close

**Comment by JmillsExpensify** [comment]
$1,000 approved for @dukenv0307
